weaviate/connect/v4.py

```diff
@@ -93,25 +97,24 @@ def __init__(
         trust_env: bool,
         additional_headers: Optional[Dict[str, Any]],
         connection_config: ConnectionConfig,
+        loop: asyncio.AbstractEventLoop,  # required for background token refresh
```

Comment:
singleton?

Reply:
This needs to be just an asyncio event loop in the case that they are running with `WeaviateAsyncClient` in which case they need to pass the running event loop to the periodic refresh token background thread

Reply:
same here, don't fully understand the reasoning

Reply:
The `_ConnectionV4` class is the fundamental implementation of the client's connection to Weaviate in both the `async` and `sync` use-cases. In both cases, the background token refreshing logic requires an open and running `asyncio.AbstractEventLoop` in order to schedule the defined coroutines on.

In the case of the `sync` parts of the codebase, this loop is managed by the `_EventLoop` class that is retrieved using the `_EventLoopSingleton` factory class. In the `async` parts of the codebase this loop already exists as the `asyncio.get_running_event_loop()`, which is called at `weaviate/client.py#146` and injected at `weaviate/client_base.py#84`.

As such, it makes sense to me that `_ConnectionV4` should depend on the `asyncio` event loop rather than on the singleton since the singleton is only relevant to the `sync` methods, which sit above `_ConnectionV4` in the structure of the codebase.

Reply:
thank you!